prow/deck/jobs/jobs.go

```diff
@@ -294,8 +299,22 @@ func (ja *JobAgent) update() error {
 			duration -= duration % time.Second // strip fractional seconds
 			nj.Duration = duration.String()
 		}
+		if j.ObjectMeta.Annotations[client.GerritID] != "" &&
+			j.ObjectMeta.Annotations[client.GerritInstance] != "" &&
+			j.ObjectMeta.Labels[client.GerritRevision] != "" {
```

Comment:
how about let's add something like a `repo-source` label? xD

Reply:
(actually, I'll need one for github as well... I can take a stab)

Reply:
Yeah, we definitely need something like that, especially if we're eventually planning to add support for pubsub, etc. I imagine we would need two fields, one for the URL and one for the type of server (Github, Gerrit, ....) But for now I just want to get this fix in without having to make any larger API changes

Reply:
Also fwiw I think this information should go in actual fields in `Refs`, not in a Label or Annotation--that's a really good way to have a bunch of magic string keys instead of actually taking of the Go type system.